### ISSUE TITLE
Add `SHOW CLUSTER REPLICAS OF <cluster>`

### ISSUE THREAD

**Issue by benesch** [MaterializeInc/materialize#15270]
### Feature request

Requested by @morsapaes on Slack ([link](https://materializeinc.slack.com/archives/C02FWJ94HME/p1664189686941239)). It'd be nice to have dedicated syntax for showing the cluster replicas of a specific cluster, much like you can show views from a specific database. Proposal:

```
SHOW CLUSTER REPLICAS OF <cluster>
```

The use of `OF` is intentional, as (IMO) you'd say "create a replica of $CLUSTER", not "create a replica for $CLUSTER".

**Comment by chaas** [comment]
We're introducing a new keyword `OF` to the `SHOW` commands by doing this, if we're ok with that.

For all the other `SHOW <object>` commands, e.g. `SHOW MATERIALIZED VIEWS` the syntax is `IN CLUSTER <cluster>`. As such, we already get `SHOW CLUSTER REPLICAS IN CLUSTER <cluster>` with barely any changes. But I also get why that doesn't make as much sense for replicas.

cc @morsapaes

**Comment by benesch** [comment]
> if we're ok with that.

Yeah, for sure. "Keywords" in SQL aren't like keywords in other programming languages, because ... they don't actually prevent you use from using that keyword as a column name, table name, etc. There's a separate class of reserved/semi-reserved keywords ...

https://github.com/MaterializeInc/materialize/blob/b43ecf8cd82a2e1db02c7b5f4344ea14873381ab/src/sql-parser/src/keywords.rs#L65-L127

that function much more like keywords in other languages. Adding keywords to that set is very scary, since it's not strictly a backcompat change.

**Comment by benesch** [comment]
Upon discussion with @chaas and @morsapaes together, an argument emerged for spelling this command `SHOW CLUSTER REPLICAS FROM` instead of `SHOW CLUSTER REPLICAS OF`. We use `FROM` whenever we want to refer to one layer up in the hierarchy (you `SHOW SCHEMAS` "from" a database, you `SHOW OBJECTS` from a schema).
